### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by romulogcerqueira**
* Optimize and improve all codes
* Unify legacy classes (CommonSonar, MultibeamSonar and ScanSonar) in a single one (Sonar)
* Unify methods: (simulateSingleBeam and simulateMultiBeam) -> simulateSonar
* Add methods: applyAdditionalGain
* Rename methods: decodeShaderImage -> convertShader, codeSonarData -> decodeShader
* Remove attributes: range, gain, bearing, start_angle, end_angle, step_angle, ping_pong_mode and reverse_scan
* Simplify Util class, removing useless code and improving the current code
* Document all code
* Update unitary test
* Remove src/test file (blank document)

### CODE REVIEW

**Comment by romulogcerqueira** [comment]
@doudou, I am opening three pull requests about the sonar simulation refactoring (`simulation/gpu_sonar_simulation`, `simulation/orogen/imaging_sonar_simulation` and `bundles/flat_fish`). 

I put `[DO NOT MERGE]` because I prefer all three pull requests be ready to merge together to avoid inconsistent packages on master.

File: src/Sonar.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+            for (double j = new_idx + 1; j < next_idx; j += 1.0)
+                dst[j] = local_slope * j + local_const;
+        }
+    }
+}
+
+float Sonar::sigmoid(float x) {
+
+	float l = 1, k = 18, x0 = 0.666666667;
```
Comment (doudou) [inline src/Sonar.cpp]: Fix the indentation and the stray empty lines while you're at it ...

**Comment by doudou** [comment]
Looks good overall.

**Comment by romulogcerqueira** [comment]
Updated code, @doudou !

**Comment by romulogcerqueira** [comment]
Indentation fixed here too!

File: src/Sonar.hpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+// Opencv includes
+#include <opencv2/core/core.hpp>
+
+namespace gpu_sonar_simulation {
+
+class Sonar {
+
+public:
+	/** Number of bins in a beam */
+	uint32_t bin_count;
```
Comment (doudou) [inline src/Sonar.hpp]: Not fixed.

**Comment by romulogcerqueira** [comment]
@doudou, could you check again?

**Comment by doudou** [comment]
Looks good now :+1:

**Comment by romulogcerqueira** [comment]
Merging here...

**Pull request merged**